### DISCUSSION TITLE
是否有自定义或新增路由功能？

### DISCUSSION THREAD

**Discussion by kecikeci** [halo-dev/halo#6554] [Q&A]
### 你当前使用的版本

2.17

### 描述一下此特性

比如我想要定义一个路由为/xxx，然后主题方提供模板页xxx.html即可，模板里提供内容即可。

这样自由度更大，可以通过Finder API随意发挥模板页xxx.html的内容。

**Comment by ruibaby** [comment]
https://docs.halo.run/developer-guide/theme/template-route-mapping#custom-templates

**Reply by kecikeci** [reply]
> https://docs.halo.run/developer-guide/theme/template-route-mapping#custom-templates

这个只能固定3种模板，且不能自由发挥内容，且会占用已有路由。

比如我想让页面同时展示 最新文章、图库、瞬间等内容，通过Finder API查询数据，然后指定一个路由就可以访问到。这种的怎么弄呢

**Reply by ruibaby** [reply]
> 且会占用已有路由

没有明白为什么会占用已有路由，其中的 page 就是支持任意路由的。

> 比如我想让页面同时展示 最新文章、图库、瞬间等内容，通过Finder API查询数据，然后指定一个路由就可以访问到。这种的怎么弄呢

推荐做法是利用 page 的自定义模板，然后在这个模板中用任意 Finder API 去查询数据，然后在 Console 创建新的页面选择这个模板。

> 比如我想要定义一个路由为/xxx，然后主题方提供模板页xxx.html即可，模板里提供内容即可。

和这里说的区别仅仅是这种方式需要去 Console 新建一个页面。

你说的这种应该是假设约定主题里面存在 page--foo.html 模板，那么就注册一个可访问的 /foo 路由。

**Reply by kecikeci** [reply]
“然后在 Console 创建新的页面选择这个模板”，区别就是在这块。这种只能是在发布文章时适用吧

**Reply by kecikeci** [reply]
你说的这种应该是假设约定主题里面存在 page--foo.html 模板，那么就注册一个可访问的 /foo 路由。

对，想要的就是这种，其实只想要一个路由

**Reply by ruibaby** [reply]
> “然后在 Console 创建新的页面选择这个模板”，区别就是在这块。这种只能是在发布文章时适用吧

<img width="1321" alt="image" src="https://github.com/user-attachments/assets/2b75c12b-8ee3-4d8c-8250-4cde366dbe9b">

页面别名就是最终访问页面的路由，并且 Halo 对页面模型的别名限制非常宽松，你甚至可以用 /about/company 这种路由。

**Reply by kecikeci** [reply]
我的需求其实是这样，我想做一个文章列表，通过Finder API 可以查到数据，但是没有路由能访问，且不想用/主页路由，也不想去自定义页面那配置。只想要一个路由能访问到模板页并展示

**Reply by ruibaby** [reply]
> 也不想去自定义页面那配置。

我没有明白这两者有什么本质区别，是交付给使用者的时候配置麻烦吗？

**Reply by kecikeci** [reply]
自定义页面，里边的内容是用不上的呀。

**Reply by ruibaby** [reply]
> 自定义页面，里边的内容是用不上的呀。

那是开发者自己的选择，可以完全不用内容。

https://demo.halo.run/about/company?preview-theme=theme-apex

**Reply by kecikeci** [reply]
理是这个理   如果能把这个路由功能单独拿出来可能会更清爽吧
